Getting-Started-With-Orleans/Developing-a-Grain.md

```diff
@@ -96,229 +51,12 @@ public class PlayerGrain : Grain, IPlayerGrain
-
-Grains that execute error-handling / recovery code _must_ catch exceptions / faulted `WriteStateAsync()` `Task`s and not re-throw to signify that they have successfully handled the write error. 
-
-## Storage Provider Framework
-
-There is a service provider API for writing additional persistence providers – `IStorageProvider`.
-
-The Persistence Provider API covers read and write operations for GrainState data.
-
-``` csharp
-public interface IStorageProvider
-{
-  Logger Log { get; }
-  Task Init();
-  Task Close();
-
-  Task ReadStateAsync(string grainType, GrainId grainId, GrainState grainState);
-  Task WriteStateAsync(string grainType, GrainId grainId, GrainState grainState);
-}
-```
-
-## Storage Provider Semantics
-
-Any attempt to perform a write operation when the storage provider detects an `Etag` constraint violation _should_ cause the write `Task` to be faulted with transient error `Orleans.InconsistentStateException` and wrapping the underlying storage exception. 
-
-``` csharp
-public class InconsistentStateException : AggregateException
-{
-  /// <summary>The Etag value currently held in persistent storage.</summary>
-  public string StoredEtag { get; private set; }
-  /// <summary>The Etag value currently held in memory, and attempting to be updated.</summary>
-  public string CurrentEtag { get; private set; }
-
-  public InconsistentStateException(
-    string errorMsg, 
-    string storedEtag, 
-    string currentEtag, 
-    Exception storageException
-    ) : base(errorMsg, storageException)
-  {
-    this.StoredEtag = storedEtag;
-    this.CurrentEtag = currentEtag;
-  }
-
-  public InconsistentStateException(string storedEtag, string currentEtag, Exception storageException)
-    : this(storageException.Message, storedEtag, currentEtag, storageException)
-  { }
-}
-```
-
-
-Any other failure conditions from a write operation _should_ cause the write `Task` to be broken with an exception containing the underlying storage exception. 
-
-## Data Mapping
-
-Individual storage providers should decide how best to store grain state – blob (various formats / serialized forms) or column-per-field are obvious choices.
+## Grain Reference
 
-The basic storage provider for Azure Table encodes state data fields into a single table column using Orleans binary serialization.
+A grain reference is a logical endpoint that allows other grains, as well as an [Orleans Client](/Orleans/Getting-Started-With-Orleans/Clients) code, to invoke methods and properties of a particular grain interface implemented by a grain. 
+A grain reference is a proxy object that implements the corresponding grain interface. 
+A grain reference can be constructed by passing the identity of the grain to the `GetGrain()` method of the factory class auto-generated at compile time for the corresponding grain interface, or receiving the return value of a method or property. 
```

Comment:
at first I thought it'd be better to remove the Grain Reference section altogether. because IMHO no one actually needs to know this internal implementation detail, because it just works. it's also a too big of a leap from grain interfaces to grain references. now that I see it's not even accurate, I'd rather remove it. what do you think?